geode-core/src/main/java/org/apache/geode/cache/configuration/RegionConfigFactory.java

```diff
@@ -0,0 +1,244 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.apache.geode.cache.configuration;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import org.apache.geode.management.internal.cli.functions.RegionFunctionArgs;
```

Comment:
Generally, I would like to see us moving away from using RegionFunctionArgs. In the command, we should start creating the RegionConfig object directly from the command arguments. This RegionFunctionArgs is there before the config objects are available. If we can completely get rid of it, that would be great.

If not, this RegionConfigFactory does not belong in the config package either. It's more used in the command to convert RegionFunctionArgs to RegionConfig, the rest of the world should only work with RegionConfig. Only the command need to be aware of this "factory" or "converter".

Reply:
I agree that the RegionConfigFactory should be in the command directory -- we will move that. With regards to the RegionFunctionArgs -- yes, we should ideally not use it, but we wanted to do this in the most incremental fashion possible. The most important takeaway here is moving away from generating XML snippets to persist, and also using SingleGfshCommand and ResultModel. Eventually we will get rid of RegionFunctionArgs.

Reply:
We'll address this in a future PR